luigi/hadoop.py

```diff
@@ -398,10 +403,17 @@ def run_job(self, job):
         cmb_cmd = '{0} mrrunner.py combiner'.format(python_executable)
         red_cmd = '{0} mrrunner.py reduce'.format(python_executable)
 
-        # replace output with a temporary work directory
         output_final = job.output().path
-        output_tmp_fn = output_final + '-temp-' + datetime.datetime.now().isoformat().replace(':', '-')
-        tmp_target = luigi.hdfs.HdfsTarget(output_tmp_fn)
+        # atomic output: replace output with a temporary work directory
+        if self.end_job_with_atomic_move_dir:
+            if isinstance(job.output(), luigi.s3.S3FlagTarget):
+                raise TypeError("end_job_with_atomic_move_dir is not supported"
```

Comment:
@mikepii I realize this PR is over a year old and you may not be able to answer my question, but do you remember why `end_job_with_atomic_move_dir=False` is required for S3?  I'm working with Hadoop on Google Cloud Platform and also had to set that flag to `False` several months ago.  I don't fully recall the reason, but i believe the output log files contained URL's that `$ hadoop fs` could not handle.  Something like with an escaped `gs://` like: `'/path/to/output/_log/something_gs%3A//'`.  The path was valid for whatever process wrote it, but not for the copy operation.  Ultimately we found the copy was too slow and relying on the `_SUCCESS` flag was fine, but is this similar to what you were seeing on AWS?

I'm working on allowing [`GCSTarget()`'s as input and output for Hadoop streaming](https://github.com/geowurster/luigi/commit/2274d905da0a33d2af4d109d5c8af01781bc0523) tasks as they are valid paths when working on the Google Cloud Platform, found this PR when doing some research, and am wondering if I should raise the same exception for `GCSFlagTarget()`'s.